ext/symengine/ruby_function.h

```diff
@@ -0,0 +1,42 @@
+#ifndef RUBY_FUNCTION_H_
+#define RUBY_FUNCTION_H_
+
+#include <ruby.h>
+#include <symengine/cwrapper.h>
+
+#include "ruby_basic.h"
+#include "symengine.h"
+#include "symengine_macros.h"
+
+VALUE cfunction_func(void (*cwfunc_ptr)(basic_struct*, const basic_struct*), VALUE operand1);
+
+VALUE cfunction_abs(VALUE self, VALUE operand1);
+VALUE cfunction_sin(VALUE self, VALUE operand1);
+VALUE cfunction_cos(VALUE self, VALUE operand1);
+VALUE cfunction_tan(VALUE self, VALUE operand1);
+VALUE cfunction_csc(VALUE self, VALUE operand1);
+VALUE cfunction_cot(VALUE self, VALUE operand1);
+VALUE cfunction_sec(VALUE self, VALUE operand1);
+VALUE cfunction_asin(VALUE self, VALUE operand1);
+VALUE cfunction_acos(VALUE self, VALUE operand1);
+VALUE cfunction_atan(VALUE self, VALUE operand1);
+VALUE cfunction_acsc(VALUE self, VALUE operand1);
+VALUE cfunction_acot(VALUE self, VALUE operand1);
+VALUE cfunction_asec(VALUE self, VALUE operand1);
+VALUE cfunction_sinh(VALUE self, VALUE operand1);
+VALUE cfunction_cosh(VALUE self, VALUE operand1);
+VALUE cfunction_tanh(VALUE self, VALUE operand1);
+VALUE cfunction_csch(VALUE self, VALUE operand1);
+VALUE cfunction_coth(VALUE self, VALUE operand1);
+VALUE cfunction_sech(VALUE self, VALUE operand1);
+VALUE cfunction_asinh(VALUE self, VALUE operand1);
+VALUE cfunction_acosh(VALUE self, VALUE operand1);
+VALUE cfunction_atanh(VALUE self, VALUE operand1);
+VALUE cfunction_acsch(VALUE self, VALUE operand1);
+VALUE cfunction_acoth(VALUE self, VALUE operand1);
+VALUE cfunction_asech(VALUE self, VALUE operand1);
+VALUE cfunction_lambertw(VALUE self, VALUE operand1);
+VALUE cfunction_dirichlet_eta(VALUE self, VALUE operand1);
+VALUE cfunction_gamma(VALUE self, VALUE operand1);
```

Comment:
Something like this https://github.com/symengine/symengine/blob/master/symengine/visitor.h#L33 can be done to avoid repetition and keeping track if all the functions have been covered or not.